api/db/changelog/db.changelog-44-notebook_cohort_cache.xml

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9
+                    http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+    <changeSet author="nehasaxena" id="db.changelog-44-cacheTable">
+        <createTable tableName="notebook_cohort_cache">
+            <column name="id" type="bigint" autoIncrement="true">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+            <column name="user_workspace_id" type="bigint">
```

Comment:
This is an interesting choice - I'm not sure how I feel about it. Semantically I'd say this cache table is more a child to the user, and references other resources - rather than being a child table of the workspace role; but it's not too big of a stretch. My main concern is that the workspace role table is not actually authoritative and I'm not sure we should add any dependencies on it: https://docs.google.com/document/d/1n8G2x0qt5o3Y5ExXcIlC77Li6CN7Xmfv83qzM-9wfkI/edit

For these reasons I think my preference would be to just put user_id and workspace_id as seperate columns here. Let me know if you had other thinking on this.

Reply:
Adding user_id and workspace _id to the table will lead to repetitive values, which is way i wanted to separate them out to another table. And while doing so i stumble upon this table and thought of re-using it.

Reply:
Not sure I follow on repetitive values. Do you mean storing the user ID+workspace multiple times? I think this should be OK.

Anyways, since user+workspace is sort of a best-effort thing per that doc, I wouldn't try to tie our infrastructure too tightly to it (workspace+user relationships may be missing from that table).

Reply:
(also if you're trying to switch to a user-based cache rather than user+workspace based cache, separating the columns will probably make this query more natural)

Reply:
It makes more sense to me that this ties to the user table. Tying it to the user-workspace table, in my opinion, overloads the intended purpose of that table.